src/EditorFeatures/Test/Utilities/BloomFilterTests.cs

```diff
@@ -111,16 +112,22 @@ public void TestSerialization()
             var stream = new MemoryStream();
             var bloomFilter = new BloomFilter(0.001, false, new[] { "Hello, World" });
 
-            using (var writer = new ObjectWriter(stream, leaveOpen: true))
+            ImmutableArray<object> keepAliveObjects;
+            using (var writer = new ObjectWriter(stream, leaveOpen: true, canKeepObjectsAlive: true))
             {
                 bloomFilter.WriteTo(writer);
+
+                keepAliveObjects = writer.TakeKeepAliveObjects();
             }
 
             stream.Position = 0;
 
             using var reader = ObjectReader.TryGetReader(stream);
             var rehydratedFilter = BloomFilter.ReadFrom(reader);
             Assert.True(bloomFilter.IsEquivalent(rehydratedFilter));
+
+            foreach (var obj in keepAliveObjects)
+                GC.KeepAlive(obj);
```

Comment:
so what's the purpose of updating this test like this? i.e. BLoomFilters don't hit these codepaths.  So is this just to get coverage of the new codepaths?

Reply:
I updated all `ObjectWriter` usages where the keep alive invariant can be maintained

Reply:
tbh i'd prefer the reverse.  we do not update this (except to pass `canKeepObjectsAlive: false`).  We then have tests running.  We woudl expect them to 100% pass forever (since bloom filters don't use keepalive).  If they fail, then that means that something is bad.  Also, if somehow bloom filters accidentally needed keep alive, then this would hopefully catch that and call it out to us.

In other words, the previous behavior of these systems was to *not* support any keepalive semantics.  So i would prefer to align with that (Esp. for those that def shoudln't have anything to do with keepalive currently).

Reply:
> Also, if somehow bloom filters accidentally needed keep alive, then this would hopefully catch that and call it out to us.

Serialization works both with and without keep alive. Objects that serialize by reference in keep alive scenarios serialize by value when keep alive is not available.

> In other words, the previous behavior of these systems was to not support any keepalive semantics.

The previous operating semantics matched what is currently represented by having keep alive semantics. The only difference is the code _assumed_ keep alive was available even when it wasn't.

Reply:
➡️ Consolidating this discussion with https://github.com/dotnet/roslyn/pull/45953#discussion_r454556159